### ISSUE TITLE
Connection with Socrata and AGOL

### ISSUE THREAD

**Issue by ShaneHunter459** [koopjs/koop-provider-agol#115]
How do I connect to a Socrata data set via AGOL using Koop? Can't I do this without downloading and installing Koop? Just looking to pass through Koop by using an end point to hit then the conversion spits out an AGOL web service.

**Comment by rgwozdz** [comment]
Hello @ShaneHunter459 .  Sorry, I'm not sure I understand your question.  Koop is not a hosted service - it is just opensource code.  You have to download, install it on a server, and configure it for your data source.  Then it's available for use in clients like AGOL.

**Comment by ShaneHunter459** [comment]
Maybe I misread this discussion but I was under the impression that I could just pass the Socrata data to Koop and it spits out a ESRI service for me to consume. See the following: https://community.esri.com/message/657017-re-is-there-a-way-to-view-socrata-data-on-an-arcgis-online-map?commentID=657017#comment-657017

**Comment by rgwozdz** [comment]
I think you have misread.  Esri just supports the development of the Koop code base  as a freely available opensource project.  Esri does not host any Koop services.  You are welcome to download, install on a server and use.  FYI, we are in the middle of upgrading the Socrata provider to work with Koop 3.x.

**Comment by ShaneHunter459** [comment]
So Koop is still alive and a viable tool for my purpose?

**Comment by rgwozdz** [comment]
Absolutely, with the caveat that the existing Socrata data adapter/provider is out-of-date, but being updated as part of the Koop road map for 2020.

**Comment by ShaneHunter459** [comment]
But right now it wouldn't work?

**Comment by rgwozdz** [comment]
No.  There is an existing Socrata provider, but it will only work with Koop 2.x.

**Comment by ShaneHunter459** [comment]
Can i still download V2?

**Comment by rgwozdz** [comment]
You can.  You would have author your Koop project from scratch and `npm install` Koop 2.x.  Honestly, I don't recommend this though.  We've put a lot of work into 3.x that fix bugs and make it more fully compatible with ArcGIS clients.

What is your timeline?  I have a rudimentary Socrata adapter for 3.x I could release;  it may work if your datasets are not that large.

**Comment by ShaneHunter459** [comment]
Right now I'm exploring just to see if this is even possible on my dataset. The dataset is not huge

**Comment by rgwozdz** [comment]
Ok, if you can give me a couple of days, I can try to get it out the door.  I assume your data has geometry; if so, what type? i.e Point, Line, etc.

**Comment by ShaneHunter459** [comment]
Ok sounds good. It is of type polygon

Thanks!

**Comment by rgwozdz** [comment]
@ShaneHunter459 - check out [https://github.com/koopjs/koop-provider-socrata](https://github.com/koopjs/koop-provider-socrata).
